### PR TITLE
Add mdbook-course to handle parsing frontmatter

### PR DESCRIPTION
**Pull request by djmitche**
In v2 of the course, I'd like to include an estimate of the time to be spent on each segment in the Markdown file. I think a good place for such metadata is in the frontmatter.

For review purposes, though, I just want to display that information. So, this is a start at a new mdbook preprocessor that just separates out the frontmatter and includes it in a `<pre>` block. Eventually, I'd like to parse it and put the time in the speaker notes.

### CODE REVIEW

**Comment by mgeisler** [comment]
Hey @djmitche, the new preprocessor seems similar to https://github.com/dvogt23/mdbook-yml-header (from [Third Party Plugins](https://github.com/rust-lang/mdBook/wiki/Third-party-plugins)), though that preprocessor seems to just throw away the front-matter.

File: mdbook-course/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# mdbook-course
+
+This is an mdBook preprocessor to handle some specific details of the
+Comprehensive-Rust course.
```
Comment (mgeisler) [inline mdbook-course/README.md:4]: I'm trying to use "Comprehensive Rust" as a noun:

```suggestion
This is an mdBook preprocessor to handle some specific details of
Comprehensive Rust.
```

File: mdbook-course/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+fn main() {
+    pretty_env_logger::init();
+    let app = Command::new("mdbook-course")
+        .about("Comprehensive-Rust-specific mdbook preprocessor")
+        .subcommand(Command::new("supports").arg(Arg::new("renderer").required(true)));
+    let matches = app.get_matches();
+
+    if let Some(_) = matches.subcommand_matches("supports") {
+        // Support all versions of mdbook.
```
Comment (mgeisler) [inline mdbook-course/src/main.rs:29]: Tiny point: This is not actually about versions, it's about the [supporting the renderer](https://rust-lang.github.io/mdBook/for_developers/preprocessors.html#hooking-into-mdbook).

File: mdbook-course/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+use clap::{Arg, Command};
+use mdbook::preprocess::CmdPreprocessor;
+use mdbook_course::frontmatter::remove_frontmatter;
+use std::io::{stdin, stdout};
+use std::process;
+
+fn main() {
+    pretty_env_logger::init();
+    let app = Command::new("mdbook-course")
+        .about("Comprehensive-Rust-specific mdbook preprocessor")
```
Comment (mgeisler) [inline mdbook-course/src/main.rs:24]:
```suggestion
        .about("mdbook preprocessor for Comprehensive Rust")
```

**Review by mgeisler — APPROVED** [review]
I'm curious to see where you can take this.

**Comment by djmitche** [comment]
> Hey @djmitche, the new preprocessor seems similar to https://github.com/dvogt23/mdbook-yml-header (from [Third Party Plugins](https://github.com/rust-lang/mdBook/wiki/Third-party-plugins)), though that preprocessor seems to just throw away the front-matter.

I looked at that crate, and yes -- it's basically the no-op preprocessor with regex-based discarding. I thought it better to use `matter` for that purpose, and I do intend to extend this implementation to actually parsing the frontmatter. So choosing `mdbook-yml-header` seemed a poor choice.

File: mdbook-course/src/main.rs (hunk 2 above)
Comment (djmitche) [reply mdbook-course/src/main.rs:29]: Oh, I was confusing this with the renderer's version-support check!

**Review by mgeisler — APPROVED** [review]
Looks good, and I like that this should be safe for the existing pages since they don't have any front-matter.

**Pull request merged**